futures-util/src/future/chain.rs

```diff
@@ -1,54 +1,52 @@
 use core::mem::PinMut;
-
-use futures_core::{Future, Poll};
-use futures_core::task;
+use futures_core::Future;
+use futures_core::task::{Context, Poll};
 
 #[must_use = "futures do nothing unless polled"]
 #[derive(Debug)]
-pub enum Chain<Fut1, Fut2, Data> {
-    First(Fut1, Option<Data>),
+crate enum Chain<Fut1, Fut2, Data> {
+    First(Option<Fut1>, Option<Data>),
```

Comment:
I can't find when this option is used, is it still needed?

Reply:
`Option<Data>` is for line 38: It is used to safely take out the data from the enum variant:
```rust
Poll::Ready(output) => (output, data.take().unwrap()),
```

`Option<Fut1>` is unnecessary. Good catch! I played with this code for quite a while and I ended up introducing  `Chain::Empty` to drop the future before the callback runs because that seemed more efficient. But it seems I messed it up. I got it right in `Chain`'s equivalent `TryChain`, though :)

I've removed it. Thanks!